File: LeetCode/java/2154_Keep_Multiplying_Found_Values_by_Two.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int findFinalValue(int[] nums, int original) {
        Set<Integer> set = new HashSet<>();

        for (int i = 0; i < nums.length; i++) {
            set.add(nums[i]);
        }

        while (true) {
            if (!set.contains(original)) {
                break;
            }

            original = original * 2;
        }

        return original;
    }
}

File: LeetCode/cpp/2154_Keep_Multiplying_Found_Values_by_Two.cpp
class Solution {
public:
	int findFinalValue(vector<int>& nums, int original) {
		set<int> st;

		for (int i = 0; i < nums.size(); i++) {
			st.insert(nums[i]);
		}

		while (1) {
			if (st.find(original) == st.end()) {
				break;
			}

			original = original * 2;
		}

		return original;
	}
};